presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlDistributedQueries.java

```diff
@@ -0,0 +1,148 @@
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.testing.AbstractTestDistributedQueries;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.testing.sql.SqlExecutor;
+import io.prestosql.testing.sql.TestTable;
+import io.prestosql.tpch.TpchTable;
+import org.testng.SkipException;
+
+import java.util.Optional;
+
+import static io.prestosql.plugin.memsql.MemSqlQueryRunner.createMemSqlQueryRunner;
+
+public class TestMemSqlDistributedQueries
+        extends AbstractTestDistributedQueries
+{
+    protected TestingMemSqlServer memSqlServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        memSqlServer = new TestingMemSqlServer();
+        closeAfterClass(() -> {
+            memSqlServer.close();
+            memSqlServer = null;
+        });
+        return createMemSqlQueryRunner(
+                memSqlServer,
+                ImmutableMap.<String, String>builder()
+                        // caching here speeds up tests highly, caching is not used in smoke tests
+                        .put("metadata.cache-ttl", "10m")
+                        .put("metadata.cache-missing", "true")
+                        .build(),
+                TpchTable.getTables());
+    }
+
+    @Override
+    protected boolean supportsDelete()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsViews()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsArrays()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsCommentOnTable()
+    {
+        return false;
+    }
+
+    @Override
+    protected boolean supportsCommentOnColumn()
+    {
+        return false;
+    }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        return new TestTable(
+                createJdbcSqlExecutor(),
+                "tpch.table",
+                "(col_required BIGINT NOT NULL," +
+                        "col_nullable BIGINT," +
+                        "col_default BIGINT DEFAULT 43," +
+                        "col_nonnull_default BIGINT NOT NULL DEFAULT 42," +
+                        "col_required2 BIGINT NOT NULL)");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getPrestoTypeName();
+        if (typeName.equals("time")
+                || typeName.equals("timestamp(3) with time zone")) {
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        if (typeName.equals("real")
+                || typeName.equals("timestamp")) {
+            // TODO this should either work or fail cleanly
+            return Optional.empty();
+        }
+
+        if (typeName.equals("varchar")) {
+            // TODO fails due to case insensitive UTF-8 comparisons
```

Comment:
You did PUSHDOWN_AND_KEEP. Does it still fail?

Reply:
Yes.

Even with `PUSHDOWN_AND_KEEP` MemSQL sees the query as

```sql
-- Presto query
select row_id from test_data_mapping_smoke_varchar_1f0ab where value >= 'łąka for the win';
-- MemSQL query
SELECT `row_id` AS `row_id` FROM `tpch`.`test_data_mapping_smoke_varchar_1f0ab` WHERE ((`value` >= 'łąka for the win'))
```

Since the comparision now happens inside MemSQL and it defaults to a collation that is case-insensitive (and not UTF-8 friendly) and hence the tests fail. I was able to get the tests to pass by changing the collation of the connection (only affects literal values, not any values picked from tables) to `binary`.